IDinsight/aaq-core
# AAQ-128: Add security to admin app and manage content endpoints

Reviewer: @ziakhan04 
Estimate: 2 hrs

----

# AAQ-128: Add security to admin app and manage content endpoints

![image](https://github.com/IDinsight/aaq-core/assets/14125957/35daf38e-146e-440d-9a94-0ac148102c1d)

## Ticket

Fixes: [AAQ-128](https://idinsight.atlassian.net/browse/AAQ-128?atlOrigin=eyJpIjoiYTdmMWYyODY1MmE4NDVhNWFmNTYxYjQ0NDkyMTEzZWMiLCJwIjoiaiJ9)

## Description, Motivation and Context

Secure admin app and content management endpoints

### Changes

**Backend**
* Add oauth bearer token
* Add login and token generation
* Add two users - "readonly" and "fullaccess"
  * "readonly" cannot edit, create, or delete content. "fullaccess" can.
* Added tests

**Admin app:**
* Added login screen
* Added logout to nav bar
* Added conditionally showing edit, add, and delete buttons based on access level
* Added token retrieval and storage
* Expired or missing token takes you to login screen

## How Has This Been Tested?
- New tests for backend
- Frontend test cases run manually
  - login as "readonly"
    - confirm can see cards and content 
    - confirm no edit/delete/add options
    - logout -> should redirect to login page
  - login as "fullaccess"
    - confirm can see cards and content
    - confirm can edit/delete/add
    - logout -> should redirect to login page
  - open root "localhost:3000/" in incognito window
    - Notice that redirected to login page

## Checklist:

This checklist is a useful reminder of small things that can easily be forgotten.
Put an `x` in all the items that apply and remove any items that are not relevant to this PR.


[1]: http://chris.beams.io/posts/git-commit/
